include/sys/thread_stack.h

```diff
@@ -61,31 +100,204 @@ struct __packed z_thread_stack_element {
+ * @brief Define a toplevel kernel stack memory region
+ *
+ * This declares a region of memory for use as a thread stack, for threads
+ * that exclusively run in supervisor mode. This is also suitable for
+ * declaring special stacks for interrupt or exception handling.
+ *
+ * Stacks declared with this macro may not host user mode threads.
+ *
+ * It is legal to precede this definition with the 'static' keyword.
+ *
+ * It is NOT legal to take the sizeof(sym) and pass that to the stackSize
+ * parameter of k_thread_create(), it may not be the same as the
+ * 'size' parameter. Use K_KERNEL_STACK_SIZEOF() instead.
+ *
+ * The total amount of memory allocated may be increased to accommodate
+ * fixed-size stack overflow guards.
+ *
+ * @param sym Thread stack symbol name
+ * @param size Size of the stack memory region
  */
-#ifdef ARCH_THREAD_STACK_DEFINE
-#define K_THREAD_STACK_DEFINE(sym, size) ARCH_THREAD_STACK_DEFINE(sym, size)
-#define K_THREAD_STACK_ARRAY_DEFINE(sym, nmemb, size) \
-		ARCH_THREAD_STACK_ARRAY_DEFINE(sym, nmemb, size)
-#define K_THREAD_STACK_LEN(size) ARCH_THREAD_STACK_LEN(size)
-#define K_THREAD_STACK_MEMBER(sym, size) ARCH_THREAD_STACK_MEMBER(sym, size)
-#define K_THREAD_STACK_SIZEOF(sym) ARCH_THREAD_STACK_SIZEOF(sym)
-#define K_THREAD_STACK_RESERVED ((size_t)ARCH_THREAD_STACK_RESERVED)
-static inline char *Z_THREAD_STACK_BUFFER(k_thread_stack_t *sym)
+#define K_KERNEL_STACK_DEFINE(sym, size) \
+	struct z_thread_stack_element __noinit \
+		__aligned(Z_KERNEL_STACK_OBJ_ALIGN) \
+		sym[Z_KERNEL_STACK_SIZE_ADJUST(size)]
+
+#define Z_KERNEL_STACK_LEN(size) \
+	ROUND_UP(Z_KERNEL_STACK_SIZE_ADJUST(size), Z_KERNEL_STACK_OBJ_ALIGN)
+
+/**
+ * @def K_KERNEL_STACK_ARRAY_DEFINE
+ * @brief Define a toplevel array of kernel stack memory regions
+ *
+ * Stacks declared with this macro may not host user mode threads.
+ *
+ * @param sym Kernel stack array symbol name
+ * @param nmemb Number of stacks to declare
+ * @param size Size of the stack memory region
+ */
+#define K_KERNEL_STACK_ARRAY_DEFINE(sym, nmemb, size) \
+	struct z_thread_stack_element __noinit \
+		__aligned(Z_KERNEL_STACK_OBJ_ALIGN) \
+		sym[nmemb][Z_KERNEL_STACK_LEN(size)]
+
+/**
+ * @def K_KERNEL_STACK_MEMBER
+ * @brief Declare an embedded stack memory region
+ *
+ * Used for kernel stacks embedded within other data structures.
+ *
+ * Stacks declared with this macro may not host user mode threads.
+ * @param sym Thread stack symbol name
+ * @param size Size of the stack memory region
+ */
+#define K_KERNEL_STACK_MEMBER(sym, size) \
+	struct z_thread_stack_element \
+		__aligned(Z_KERNEL_STACK_OBJ_ALIGN) \
+		sym[Z_KERNEL_STACK_SIZE_ADJUST(size)]
+
+#define K_KERNEL_STACK_SIZEOF(sym) (sizeof(sym) - K_KERNEL_STACK_RESERVED)
+
+static inline char *Z_KERNEL_STACK_BUFFER(k_thread_stack_t *sym)
 {
-	return ARCH_THREAD_STACK_BUFFER(sym);
+	return (char *)sym + K_KERNEL_STACK_RESERVED;
 }
+#ifndef CONFIG_USERSPACE
+#define K_THREAD_STACK_RESERVED		K_KERNEL_STACK_RESERVED
```

Comment:
fix indentation

Reply:
I see no problem with indentation in my text editor (VS Code) can you elaborate.

Reply:
https://imgur.com/a/oCTMJvY

Reply:
@ioannisg, I think this is a side affect of github defaulting the tabs to 4 and @andrewboie using tabs between words. There is some sort of addon you can use to get this github display to use 8 for tabs or Andrew could use spaces between words. I'm not sure though what our general style guide is on this part.

Reply:
Yeap, if so, please disregard :)